numalogic/connectors/druid.py

```diff
@@ -50,13 +51,17 @@ def fetch_data(
         start_dt = end_dt - timedelta(hours=hours)
         intervals = f"{start_dt.isoformat()}/{end_dt.isoformat()}"
 
+        dimension_specs = []
+        for d in dimensions:
+            dimension_specs.append(DimensionSpec(dimension=d, output_name=d))
```

Comment:
```suggestion
        dimension_specs = map(lambda d: DimensionSpec(dimension=d, output_name=d), dimensions)
```
Can you see if this works? This basically creates a generator object instead of a list.

If not, you can do:
`dimension_specs = [DimensionSpec(dimension=d, output_name=d) for d in dimensions]`